libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackMetadata.kt

```diff
@@ -0,0 +1,197 @@
+package com.mapbox.navigation.core.telemetry.events
+
+import android.location.Location
+import com.google.gson.Gson
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.utils.toTelemetryLocations
+import com.mapbox.navigation.core.telemetry.LocationsCollector
+
+/**
+ * It's the wrapper of [FeedbackMetadata] that collect locations after instance of
+ * [FeedbackMetadataWrapper] is created and stop collecting in 2 possible cases:
+ * - locations buffer is full;
+ * - [FeedbackMetadataWrapper.get] is called.
+ *
+ * Notes:
+ * - if you need to serialize Feedback metadata, use for that [FeedbackMetadata] only (see [FeedbackMetadataWrapper.get]);
+ * - to collect as much data as possible preferably call [FeedbackMetadataWrapper.get] just before posting or storing feedback;
+ * - [FeedbackMetadataWrapper.get] can be invoked whenever it needs.
```

Comment:
I'd remove this line, it's kinda self-implied that you can unwrap whenever needed.